lightning/src/ln/interactivetxs.rs

```diff
@@ -353,23 +341,25 @@ impl ConstructedTransaction {
 	}
 
 	pub fn compute_txid(&self) -> Txid {
-		self.build_unsigned_tx().compute_txid()
+		self.tx().compute_txid()
 	}
 
 	/// Adds provided holder witnesses to holder inputs of unsigned transaction.
 	///
 	/// Note that it is assumed that the witness count equals the holder input count.
 	fn add_local_witnesses(&mut self, witnesses: Vec<Witness>) {
-		self.inputs
+		self.tx
+			.input
 			.iter_mut()
+			.zip(self.inputs.iter())
 			.enumerate()
-			.filter(|(_, input)| input.is_local(self.holder_is_initiator))
+			.filter(|(_, (_, input))| input.is_local(self.holder_is_initiator))
 			.filter(|(index, _)| {
 				self.shared_input_index
 					.map(|shared_index| *index != shared_index as usize)
 					.unwrap_or(true)
 			})
-			.map(|(_, input)| &mut input.txin)
+			.map(|(_, (txin, _))| txin)
 			.zip(witnesses)
 			.for_each(|(input, witness)| input.witness = witness);
 	}
```

Comment:
Wondering if we should have `InteractiveTxSigningSession` store both local and remote signatures until `finalize_funding_tx` is called. That way, `ConstructedTransaction` would always contain an unsigned transaction instead of one that may be partially signed.